chocolatey/choco
# (#2462) Add deprecation notices for list with remote sources before making it local only

## Description Of Changes

This pull request deprecates the argument local only and the arguments only relevant when calling remote sources when the list command is used.

Additionnaly it also adds several deprecation noticed both in the help page, as well as when running the list command to ensure that users are truly notified of the change to make it local only in v2.0.0.

## Motivation and Context

To make the list command only handle local/installed packages, and to notify users to the best of our ability that this change will happen.

## Testing

1. Run build.bat to ensure that the build and unit tests has not been broken
2. Run `choco list chocolatey` and ensure that there is a warning mentioning the list command will be made local only
3. Run `choco list chocolatey --noop` and ensure the same as step 2.
4. Run `choco list chocolatey --local-only` and ensure there is no deprecation notice displayed.
5. Run `choco list chocolatey --local-only --noop` and ensure the same as step 4.
6. Run `choco list --help` and ensure there is a correct deprecate notice at the top, each remote argument + local only argument is mentioned with a deprecation notice and that the alternative sources will be removed in the list command.
7. Run `choco search chocolatey` and ensure there is no deprecation notice displayed.
8. Run `choco search chocolatey --noop` and ensure the same as step 7.
9. Run `choco search chocolatey --local-only` and ensure the same as step 7.
10. Run `choco search chocolatey --local-only --noop` and ensure the same as step 7.
11. Run `choco search --help` and ensure the only deprecation notice is for the list example.
12. Repeat step 7-11 using the `find` command.


## Change Types Made



## Related Issue


Fixes #2462

## Change Checklist
